.github/workflows/verify-dashboards-alerts.yaml

```diff
@@ -0,0 +1,135 @@
+
+          sudo apt-get update -y
+          sudo apt-get install -y golang expect
+
+          export ISTIO_INSTALL_SAIL=true
+
+          cat << 'EOF' > automate_setup.expect
+          #!/usr/bin/expect -f
+          set timeout -1
+          spawn bash hack/quickstart-setup.sh
+          expect "Are you ready to begin? (y/n)"
+          send "y\r"
+          expect "Do you want to set up a DNS provider for use with Kuadrant's DNSPolicy API? (y/n)"
+          send "n\r"
+          expect "Thank you for using Kuadrant! If you have any questions or feedback, please reach out to our community."
+          EOF
+          chmod +x automate_setup.expect
+          ./automate_setup.expect
+
+      - name: Port forward grafana and check if contains dashboards.
+        run: |
+          # Port forward Grafana
+          kubectl -n monitoring wait --for=condition=available deployment grafana --timeout=600s
+          kubectl -n monitoring port-forward service/grafana 3000:3000 &
+          grafana_process_id=$!
+          echo "Successfully port forwarded Grafana service."
+
+          sleep 1
+
+          # Make API Call and save response to variable.
+          grafana_api_call=$(curl -u admin:admin http://127.0.0.1:3000/api/search)
+
+          # Compare the content in json file with field containing dashboard names
+
+          app_developer=$(jq -r '.panels[1].title' examples/dashboards/app_developer.json)
+          business_user=$(jq -r '.panels[1].title' examples/dashboards/business_user.json)
+          platform_engineer=$(jq -r '.panels[1].title' examples/dashboards/platform_engineer.json)
+
+          declare -a missing_dashboards=()
+
+          if [[ "$grafana_api_call" != *"$app_developer"* ]]; then
+            echo "Grafana does not have $app_developer dashboard."
+            missing_dashboards+=("$app_developer")
+          fi
+          if [[ "$grafana_api_call" != *"$business_user"* ]]; then
+            echo "Grafana does not have $business_user dashboard."
+            missing_dashboards+=("$business_user")
+          fi
+          if [[ "$grafana_api_call" != *"$platform_engineer"* ]]; then
+            echo "Grafana does not have $platform_engineer dashboard."
+            missing_dashboards+=("$platform_engineer")
+          fi
+
+          if [[ ${#missing_dashboards[@]} -gt 0 ]]; then
+            echo "Grafana is missing the following dashboards:"
+            printf '%s\n' "${missing_dashboards[@]}"
+            echo "Exiting..."
+            exit 1
+          fi
+
+          echo "Grafana contains dashboards $app_developer, $business_user and $platform_engineer. Continuing to Prometheus..."
+
+          # Close Grafana port-forward.
+          kill $grafana_process_id
+          echo "Stoppped port forwarding Grafana."
+
+      - name: Port forward Prometheus and check if contains alert rules.
+        run: |
+          kubectl -n monitoring wait --for=condition=ready pod prometheus-k8s-0 --timeout=600s
```

Comment:
I would separete setup and test phases

Reply:
Makes sense. Will implement now.

Reply:
https://github.com/Kuadrant/kuadrant-operator/pull/708#discussion_r1668397947